src/write.jl

```diff
@@ -119,36 +147,27 @@ end
 
 function _write_constraint(
     io::IO,
-    model,
     variables,
-    F::Type{MOI.VectorOfVariables},
-    S::Type{
-        <:Union{
-            MOI.AllDifferent,
-            MOI.CountDistinct,
-            MOI.CountGreaterThan,
-            MOI.Cumulative,
-            MOI.Circuit,
-        },
+    f::MOI.VectorOfVariables,
```

Comment:
Can we split the change from `::Type{F}` to `::F` into separate PRs? The diff is quite large and most is unrelated to `ScalarNonlinearfunction`

Reply:
see #24